packages/web/src/pages/sign-up-page/components/EmailField.tsx

```diff
@@ -38,25 +46,46 @@ export const EmailField = (props: Partial<HarmonyTextFieldProps>) => {
 }
 
 export const NewEmailField = () => {
-  const [, { error }] = useField('email')
+  const dispatch = useDispatch()
+  const [{ value: email }, { error }] = useField('email')
   const { isValidating } = useFormikContext()
   const emailInUse = error === emailSchemaMessages.emailInUse
+  const isGuest = error === emailSchemaMessages.completeYourProfile
+  dispatch(setField('isGuest', isGuest))
+  const hasError = emailInUse || isGuest
   // Used to ensure the hint doesn't go away while validation is ocurring
-  const hadError = usePrevious(emailInUse)
+  const hadError = usePrevious(hasError)
 
   const signInLink = (
     <TextLink variant='visible' asChild>
       <Link to={SIGN_IN_PAGE}>{createEmailPageMessages.signIn}</Link>
     </TextLink>
   )
+  const handleClickConfirmEmail = useCallback(() => {
+    dispatch(setValueField('email', email))
+    dispatch(setValueField('password', TEMPORARY_PASSWORD))
+    dispatch(signIn(email, TEMPORARY_PASSWORD))
+  }, [dispatch, email])
+
+  const confirmEmailLink = (
+    <TextLink variant='visible' asChild>
```

Comment:
use the TextLink from web which wraps harmony text-link and react-router link together.